pkg/graveler/ref/resolve_ref.go

```diff
@@ -26,7 +26,33 @@ func isAHash(part string) bool {
 
 // revResolve return the first resolve of 'rev' - by hash, branch or tag
 func revResolve(ctx context.Context, store Store, addressProvider ident.AddressProvider, repository *graveler.RepositoryRecord, rev string) (*graveler.ResolvedRef, error) {
-	resolvers := []revResolverFunc{revResolveAHash, revResolveBranch, revResolveTag}
+	// looking for commit first - a full commit takes precedence over a branch or a tag
+	r, err := revResolveAHash(ctx, store, addressProvider, repository, rev)
```

Comment:
Can we keep one set of func and just split the `revResolveAHash` to `revResolveAHash` and `revResolveAHashPrefix`?
This way we can place `revResolveAHash` first and `revResolveAHashPrefix` last.